include/deal.II/base/mpi.h

```diff
@@ -371,14 +379,35 @@ namespace Utilities
      * array is the minimum over the i-th entries of the input arrays from each
      * processor.
      *
+     * This function can also be used for ArrayView objects initialized with
+     * data living in CUDA memory if MemorySpace::CUDA is chosen.
+     *
      * Input and output arrays may be the same.
      */
-    template <typename T>
+    template <typename T, typename MemorySpaceType = dealii::MemorySpace::Host>
     void
     min(const ArrayView<const T> &values,
         const MPI_Comm &          mpi_communicator,
         const ArrayView<T> &      minima);
 
+    template <typename T, typename MemorySpaceType = dealii::MemorySpace::Host>
+    void
+    Isend(const ArrayView<const T> &values,
```

Comment:
documentation missing; same for the next function

Reply:
Yes, that is all WIP.